presto-record-decoder/src/main/java/io/prestosql/decoder/raw/RawColumnDecoder.java

```diff
@@ -213,14 +213,16 @@ public FieldValueProvider decodeField(byte[] value)
         private final String columnName;
         private final Type columnType;
         private final int size;
+        private final int start;
 
-        public RawValueProvider(ByteBuffer value, FieldType fieldType, String columnName, Type columnType)
+        public RawValueProvider(ByteBuffer value, FieldType fieldType, String columnName, Type columnType, int start)
```

Comment:
You don't really need the new argument, you could get it as `value.position()`.

Reply:
from the bugfix perspective, it's fine either way

from code perspective, it could be beneficent to switch to `Slice` (optional; as a second commit, or follow-up, or not at all)